Nim/Program.cs

```diff
@@ -0,0 +1,441 @@
+                        exitProgram = true;
+                        break;
+                    default:
+                        Console.WriteLine("Invalid choice! Please select one of the appropriate options.");
+                        break;
+                }
+            }
+        }
+        static void BasicNim()
+        {
+            bool backToMenu = false;
+            while (!backToMenu) {
+                Console.WriteLine();
+                Console.WriteLine("You're inside of Nim Game (basic)!");
+                Console.WriteLine("Press 'C' to Confirm. Press 'Q' to go to Main Menu.");
+                Console.WriteLine($@"You've won {_winCountBasicMenu} times. You've lost {_lostCountBasicMenu} times.");
+
+                string menuSelection = Console.ReadLine()!.ToUpper();
+                if (menuSelection == "C") {
+                    //If instead of switch for fun
+                    Console.WriteLine();
+                    Console.WriteLine("Welcome to Nim!");
+                    int matchesCount = 24;
+                    int playerMove = 0;
+                    bool playerTurn = false;
+                    bool aiTurn = false;
+                    bool gameOver = false;
+                    Random ai = new Random();
+
+                    nim:
+                    while (!gameOver) {
+                        DisplayMatches(matchesCount);
+                        Console.WriteLine("How many matches would you like to draw? (1-3)");
+                        playerTurn:
+                        playerTurn = true;
+                        aiTurn = false;
+                        string input = Console.ReadLine();
+                        if (!Int32.TryParse(input, out playerMove)) {
+                            //TryParse. If Parsing to Int not possible, Invalid input
+                            Console.WriteLine("Invalid input! Please enter an integer.");
+                            goto playerTurn;
+                        }
+                        
+                        switch (playerMove) {
+                            case 1:
+                            case 2:
+                            case 3: //case 1-2-3 do the same things
+                                if (playerMove > matchesCount) {
+                                    Console.WriteLine($"You can only draw up to {matchesCount} matches.");
+                                    goto playerTurn;
+                                }
+
+                                matchesCount -= playerMove; //update
+                                break;
+                            default:
+                                Console.WriteLine("Please enter a valid amount! (1-3)");
+                                goto playerTurn;
+                        }
+
+                        DisplayMatches(matchesCount); //End of player turn
+                        if (matchesCount <= 0 && playerTurn == true) {
+                            gameOver = true;
```

Comment:
I think you could do a simpler main loop by using either more functions or introducing more goto labels instead of so many flags. Also, `playerTurn` and `aiTurn` seem to be redundant variables. You only need either of bot.

```c#
PlayerTurn:
// player logic
if(invalidInput) goto PlayerTurn;
matches -= ...;
if(matches == 0) goto GameOver;

// ai logic
matches -=...;
if(matches == 0) goto GameOver;
goto PlayerTurn;
GameOver:
   // print gameOver
```